addons/osfstorage/views.py

```diff
@@ -322,6 +326,10 @@ def osfstorage_create_child(file_node, payload, **kwargs):
                     ),
                     dict(payload['metadata'], **payload['hashes'])
                 )
+                is_node = file_node.target_content_type.model == 'abstractnode'
+                if version.identifier != old_version_number and is_node and not file_node.target.is_quickfiles:
+                    enqueue_postcommit_task(update_storage_usage_cache, (file_node.target._id,), {}, celery=True)
+
```

Comment:
I don't think we're covering all of our bases here for when we need to update the cache - specifically, move/copy operations.

For example, what about if I add a file, create a folder, and then make a copy of that file into that folder? This is not currently updating the cache, however, if I upload an identical copy of the same file into both the root, and a subfolder, the cache is updated. (I don't know the specifics of how storage works here - is a copy a literal duplicate, or is it a pointer to the same item?)

Another example, I add a file, add a component, and then move my file into the component.  My project won't have any files, but the cache won't have been updated.

Reply:
> (I don't know the specifics of how storage works here - is a copy a literal duplicate, or is it a pointer to the same item?)

So here's the thing and I think this deserves more discussion at the Product level. Yes all exact copies are basically pointers... All of them if a 100 users upload the same 100GBs to 100 different nodes times, it still only takes up a 100GB of storage, so this number isn't strictly accurate. Our Google Cloud implementation does this for us.

Reply:
ah okay!  My opinion is that we should treat copies as if they are literal duplicates (where displaying storage usage is concerned). If I copy a file into a folder, that's file X 2.  If I upload an identical file that is already in another folder, that's also the file X2.  I wouldn't want to add other calls to figure out if the file exists, and also, with copies, who takes the storage hit?  If we both have a copy of the same file, does it count against me or against you?

But yes, this is just my first thought, and looping Product in is a good idea.